### ISSUE TITLE
auth0_connection : import state error for google-oauth2

### ISSUE THREAD

**Issue by arrase** [auth0/terraform-provider-auth0#316]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

```
$ terraform import auth0_connection.google_oauth2 XXXXXXXXXXXXXXXXX
auth0_connection.google_oauth2: Importing from ID "XXXXXXXXXXXXXXXXX"...
auth0_connection.google_oauth2: Import prepared!
  Prepared auth0_connection for import
auth0_connection.google_oauth2: Refreshing state... [id=XXXXXXXXXXXXXXXXX]
╷
│ Error: decoding response payload failed: json: cannot unmarshal string into Go struct field ConnectionOptionsGoogleOAuth2.allowed_audiences of type []interface {}
│
│
╵
```

### Expectation

Import the current state from google-oauth2 connection kind

### Reproduction

1. Configure a google-oauth2 connection with your own client_id and client_secret
2. Try to import the state from terraform with:
```
terraform import auth0_connection.google_oauth2 XXXXXXXXXXXXXXXXX
```

### Auth0 Terraform Provider version

0.37.0

### Terraform version

1.2.6

**Comment by jmisur** [comment]
Related to https://github.com/auth0/go-auth0/issues/14

**Comment by sergiught** [comment]
Hey folks 👋🏻 , thanks for reporting this!

This is related to auth0/go-auth0#14 indeed. When the google oauth2 connection gets first created it will return the `allowed_audiences` as an empty string instead of an empty array. 

Not ideal, but until we fix this on the go auth0 SDK side a workaround would be to fill up the field in question in the UI with some dummy value, save, clear, save again. This will make it such that the API will return the field as an array afterwards and not fail any more when importing.

**Comment by mswezey23** [comment]
I'm running into this issue as of now after configuring non-dev Google API keys. 
This issue doesn't exist in our `alpha` tenant which still uses Auth0's default dev keys.

**Comment by renan-abm** [comment]
FYI: this also affects the new `auth0_connection_client` resource

```
│ Error: failed to unmarshal response payload: json: cannot unmarshal string into Go struct field ConnectionOptionsGoogleOAuth2.allowed_audiences of type []string
│ 
│   with auth0_connection_client.app["google"],
│   on auth.tf line 63, in resource "auth0_connection_client" "app":
│   63: resource "auth0_connection_client" "app" {
```

**Comment by Jno21** [comment]
Hi, 

Same issue here and it is quite important on our side. Any way we can help to solve this issue ?

**Comment by mjpolak** [comment]
+1

**Comment by sergiught** [comment]
Hey folks 👋🏻 

Apologies for the wait on this one, we were trying to push for a patch for this on the API side, but unfortunately that won't be possible so we're gonna have to work around this within the Go SDK that powers this provider (https://github.com/auth0/go-auth0). We're aiming at having this ready for the provider some time next week. I'll keep you updated in case of delays. 

Thanks!

**Comment by mjpolak** [comment]
@sergiught Thanks for feedback!

**Comment by mjpolak** [comment]
Hey @sergiught any chance for release today?

**Comment by sergiught** [comment]
Hey folks, appreciate everyone's patience with this one. 

The fix is available now in the latest release [v0.45.0](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.45.0).  Please give it a go and let us know if you have any issues with it. Thanks!

Docs: https://registry.terraform.io/providers/auth0/auth0/0.45.0/docs

**Comment by mjpolak** [comment]
Thanks, it works!
